ui/tests/integration/components/sync/secrets/page/overview-test.js

```diff
@@ -30,357 +30,344 @@ module('Integration | Component | sync | Page::Overview', function (hooks) {
+    assert.dom(updated(0)).hasText(updatedDate, 'Last updated datetime renders');
 
-  module('secrets sync is activated', function () {
-    test('it should hide the opt-in banner', async function (assert) {
-      await this.renderComponent();
+    assert.dom(total(1)).hasText('0', '# of external secrets renders for destination with no associations');
+    assert
+      .dom(updated(1))
+      .hasText('—', 'Last updated placeholder renders for destination with no associations');
 
-      assert.dom(overview.optInBanner.container).doesNotExist();
-    });
+    await click(actionToggle(0));
+    assert.dom(action('sync')).hasText('Sync secrets', 'Sync action renders');
+    assert.dom(action('details')).hasText('View synced secrets', 'View synced secrets action renders');
   });
 
-  module('with no destinations', function (hooks) {
-    hooks.beforeEach(function () {
-      this.destinations = [];
-    });
-
-    test('it should show a CTA', async function (assert) {
-      await this.renderComponent();
+  test('it should paginate the table', async function (assert) {
+    this.setup({ isActivated: true, isEnterprise: true, hasFeature: true, hasDestinations: true });
+    await this.renderComponent();
 
-      assert.dom(cta.button).exists();
+    const { name, row } = overview.table;
+    assert.dom(row).exists({ count: 3 }, 'Correct number of table rows render based on page size');
+    assert.dom(name(0)).hasText('destination-aws', 'First destination renders on page 1');
 
-      assert
-        .dom(overview.createDestination)
-        .doesNotExist('create new destination link is hidden if there are no pre-existing destinations');
-    });
+    await click(pagination.next);
+    await settled();
+    assert.dom(overview.table.row).exists({ count: 3 }, 'New items are fetched and rendered on page change');
+    assert.dom(name(0)).hasText('destination-gcp', 'First destination renders on page 2');
+  });
 
-    test('it should hide the overview cards', async function (assert) {
-      await this.renderComponent();
-
-      assert
-        .dom(overviewCard.title('Secrets by destination'))
-        .doesNotExist('it does not show secrets by destination card if there are no destinations');
-      assert
-        .dom(overviewCard.title('Total destinations'))
-        .doesNotExist('it does not show total destinations card if there are no destinations');
-      assert
-        .dom(overviewCard.title('Total secrets'))
-        .doesNotExist('it does not show total secrets card if there are no destinations');
+  test('it should show an empty state if there is an error fetching associations', async function (assert) {
+    this.setup({ isActivated: true, isEnterprise: true, hasFeature: true, hasDestinations: true });
+    this.server.get('/sys/sync/destinations/:type/:name/associations', () => {
+      return new Response(403, {}, { errors: ['Permission denied'] });
     });
-  });
 
-  module('with destinations', function () {
```

Comment:
removed a lot of the nesting modules here to get ahead of this deprecation warning: 
```
The `beforeEach` hook was called inside the wrong module (`Integration | Component | sync | 
Page::Overview > with destinations`). Instead, use hooks provided by the callback to the 
containing module (`Integration | Component | sync | Page::Overview`). 
This will become an error in QUnit 3.0.